libs/x-content/src/test/java/org/opensearch/common/xcontent/XContentParserTests.java

```diff
@@ -85,7 +85,8 @@ public class XContentParserTests extends OpenSearchTestCase {
         () -> randomAlphaOfLengthBetween(1, SmileXContent.DEFAULT_MAX_STRING_LEN),
         /* YAML parser limitation */
         XContentType.YAML,
-        () -> randomRealisticUnicodeOfCodepointLengthBetween(1, YamlXContent.DEFAULT_CODEPOINT_LIMIT)
+        /* use 75% of the limit, difficult to get the exact size of the content right */
+        () -> randomRealisticUnicodeOfCodepointLengthBetween(1, (int) (YamlXContent.DEFAULT_CODEPOINT_LIMIT * 0.75))
```

Comment:
Rather than update this individual test, lets fix the generation function.  The `limit`, at least in snake yaml is the max number of bytes, can we generate the text from the library and then truncate extra bytes?

Reply:
> Rather than update this individual test, lets fix the generation function. The `limit`, at least in snake yaml is the max number of bytes, can we generate the text from the library and then truncate extra bytes?

So the test generates the value of the string field, and the limit is expressed not in bytes but codepoints. SnakeYaml applies the limit to whole document which has some other random parts, that what makes test unstable.

Reply:
Yuck!  This is good enough to keep the test from failing